spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClientAutoConfiguration.java

```diff
@@ -100,4 +92,21 @@ public KubernetesDiscoveryProperties getKubernetesDiscoveryProperties() {
 		return new KubernetesDiscoveryProperties();
 	}
 
+	@Configuration
+	@ConditionalOnBlockingDiscoveryEnabled
+	@ConditionalOnKubernetesDiscoveryEnabled
+	public static class KubernetesDiscoveryClientConfiguration {
```

Comment:
Why a nested class here?

Reply:
The reactive service discovery needs the other beans too, and I didn't want to break backwards compatibility in case someone is doing weird stuff by disabling auto configuration and importing `KubernetesDiscoveryClientAutoConfiguration `.